src/api/config.c

```diff
@@ -612,6 +612,53 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr, int ParamMaxLength)
+{
+    FILE *fPtr;
+    int foundSection = 0;
+    if (ParamPtr == NULL || SectionName == NULL || ParamName == NULL)
+        return M64ERR_INPUT_INVALID;
+    if (FileName == NULL || (fPtr = fopen(FileName, "rb")) == NULL)
+    {
+        DebugMessage(M64MSG_ERROR, "Unable to open config file '%s'.", FileName);
+        return M64ERR_INPUT_INVALID;
+    }
+
+    char *buffer = malloc(ParamMaxLength + 64); //ParamMaxLength is the size of the value, we add 64 to make sure we have enough room to store the rest of the line.
+
+    while (fgets(buffer, sizeof(buffer), fPtr) != NULL)
```

Comment:
You need to replace ```sizeof(buffer)``` by ```(ParamMaxLength + 64)``` and I still don't understand why you're adding 64. 

Anyway I was checking how ```ini_parse_line``` was used in the rest of the code and I like the way it's used in ```ConfigInit```. Instead of reading the file line by line, the entire file is read into a dynamically allocated buffer (```sizeof(file length)```). The buffer is than parsed from beginning to end using the ```ini_parse_line``` function.

Sorry for asking you to amend this commit again and again :-(